rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/api/RecoveryCoordinator.java

```diff
@@ -152,4 +153,41 @@ public String replaceCompensator(
     public List<LRAStatusHolder> getRecoveringLRAs() {
         return lraService.getAllRecovering(true);
     }
+
+    @GET
+    @Path("failed")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "List failed Long Running Actions",
+            description = "Returns LRAs that have failed. " +
+                    " Failure records are vital pieces of data needed to aid failure tracking and analysis " +
+                    " and are retained for inspection.")
+    @APIResponse(responseCode = "200",
+            content = @Content(schema = @Schema(type = SchemaType.ARRAY, implementation = LRAStatusHolder.class)))
+    public List<LRAStatusHolder> getFailedLRAs() {
+        return lraService.getFailedLRAs();
+    }
+
+    @DELETE
+    @Path("{LraId}")
+    @Operation(summary = "Remove the log for an LRA")
+    @APIResponses({
+            @APIResponse(responseCode = "500",
+                    description = "The LRA was not recognised"),
+            @APIResponse(responseCode = "200",
+                    description = "If the LRA was successfully removed",
+                    content = @Content(schema = @Schema(implementation = Boolean.class)))
+    })
+    public Response deleteLRA(
+            @Parameter(name = "LraId", description = "The unique identifier of the LRA", required = true)
+            @PathParam("LraId")String lraId) throws NotFoundException {
+        try {
+            if (lraService.removeLog(new URI(lraId))) {
+                return Response.ok().build();
+            }
+
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        } catch (URISyntaxException e) {
+            return Response.status(INTERNAL_SERVER_ERROR).entity(String.format("%s: %s", lraId, e.getMessage())).build();
```

Comment:
I think it would be good to log the full exception stacktrace when asked by more verbose logging.
It's handy in ivestigation on what's happening in the system.
I think when `TRACE` level is enabled the exception with stacktrace and probably some information what was the reason should be logged in to the coordinator log.

Reply:
+1

Reply:
Although this stack trace will only be showing Resteasy internals since the exception is saying that the user passed an invalid URI and therefore the stacktrace will not provide any useful information (just noise).

Reply:
I think information that the uri syntax was wrong with details why it was wrong could be still usable for a detailed analysis.

Reply:
Oh yes definitely. All I am saying is that the stacktrace will not provide any additional information. I am adding trace level stack traces for the other places you ask for though.

Reply:
Fixed with the last commit